src/gui.cpp

```diff
@@ -502,10 +502,32 @@ bool window_update_controller::check_redraw(GtkWidget *toplevel)
 gui_environment::gui_environment()
 {
     keyfile = g_key_file_new();
+
+    string filename;
     
-    gchar *fn = g_build_filename(getenv("HOME"), ".calfrc", NULL);
-    string filename = fn;
-    g_free(fn);
+    gchar *fn_under_home = g_build_filename(getenv("HOME"), ".calfrc", NULL);
+    char *xdg_conf = getenv("XDG_CONFIG_HOME");
+    gchar *calf_conf_dir = g_build_filename(xdg_conf, "calf", NULL);
+    gchar *fn_under_conf = g_build_filename(calf_conf_dir, "calfrc", NULL);
+
+    //if $HOME/.calfrc exists or XDG_CONFIG_HOME is not set use $HOME/.calfrc
+    if(g_file_test(fn_under_home, G_FILE_TEST_IS_REGULAR) || xdg_conf == NULL) {
+        filename = fn_under_home;
+    } else {
+        //if $XDG_CONFIG_HOME/calf does not exist, create it
+        if(!g_file_test(calf_conf_dir, G_FILE_TEST_EXISTS))
+            g_mkdir_with_parents(calf_conf_dir, 0644);
```

Comment:
It might be nit-picking, but in my `.config/`, all subdirectories have either modes 0755 or 0700. If you give read access without "x", then it means users can only access contents of the directory if they know the file names. Might be OK in this case, I just wonder if you really intended that.

Reply:
honestly no, I didn't know this was the case. should i change it to 0755?

Reply:
Oh yes, that would be really nice. Thanks.